tsconfig.esm.json

```diff
@@ -6,6 +6,10 @@
     "outDir": "./lib/esm",
     "sourceMap": false,
     "declaration": true,
-    "declarationMap": true
-  }
+    "declarationMap": true,
+    "declarationDir": "./lib"
```

Comment:
If this value is the same that the base config (cjs) output, it should be in that.

Reply:
The base config is *tsconfig.json* now (`"extends": "./tsconfig.json"`).

I think this way the configuration is more explicit, that is, more easy to follow.

Reply:
I can not see it in the CJS config, is it intentional?

Reply:
Both *tsconfig.cjs.json* and *tsconfig.esm.json* extend *tsconfig.json* now. However, CJS files are emitted into the *lib* directory (along with typings) and ESM files into *lib/esm*.

I did this to keep the library as backward compatible as possible, though, as I see, the standard configuration nowadays is to emit CJS into *lib/cjs*, ESM into *lib/esm* and typings into *lib/types*. (For example, React SDK is configured this way.)

One more thing to mention: the CJS and ESM build outputs exactly the same typings. This is way I configured them to output the typings into the same directory (to remove unnecessary redundancy).

Reply:
Same typings, same output...for me it means that it should be in the base. I know it processes twice, but I do not think that it is an issue.